### PR TITLE
Add Lines support to the `case edit` tab.

### PR DESCRIPTION
**Pull request by Aritra8438**
This PR enables user to add lines to the cases and modify them.

https://github.com/omegaup/omegaup/assets/64671908/531283b8-6cd2-48d7-8127-d96bac1e6ac3

# Comments
-  Currently, the problem is, when lines are dragged, store and getters are working perfectly, but the reflection is not showing up.
# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by Aritra8438** [comment]
https://github.com/omegaup/omegaup/assets/64671908/efb242c1-468f-4c8a-81fb-30e597757d2a


https://github.com/omegaup/omegaup/assets/64671908/6a4b0945-92d4-4893-849b-769e1d63a523

@pabo99, @carlosabcs PTAL! It's draft and I haven't added tests.
I raised it as I couldn't fix this behaviour.

So, when I dragged `a` to `b`, store and getters are getting updated but it's not showing up in the DOM. When I am going to other case and coming back again, it's reflected.

**Comment by carlosabcs** [comment]
@pabo99 could you please give us a hand with this bug?

**Comment by Aritra8438** [comment]
I have removed the `sortable` directive for now, PTAL! @carlosabcs.

All the other PRs are blocked for this, so going to add the `sortable` part in another PR.
So, didn't remove the functions necessary for that.

File: frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue [hunk 1]
```diff
@@ -41,23 +41,185 @@
...
+                    <b-form-input
+                      v-model="line.label"
+                      size="sm"
+                      :placeholder="T.problemCreatorLabelPlaceHolder"
+                    />
+                  </b-col>
+                  <b-col cols="5" class="pr-0 text-center">
+                    <b-form-input
+                      v-if="
+                        line.data.kind === 'line' || line.data.kind === 'array'
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue:71]: Can we move this and the other conditions to a computed property? That way, we can create an `enum` and use it to determine what to display, and remove the logic from the template, just checking the value from the computed property.

For example:
```ts
enum TestEnum {
  OPTION_A = 'Option_A',
  OPTION_B = 'Option_B',
}
get myComputedProp() {
  if (kind === 'line' || kind === 'array') {
    return TestEnum.OPTION_A;
  }
  return TestEnum.OPTION_B;
}

<b-form-textarea v-if="myComputedProp === TestEnum.OPTION_A" />
```

File: frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue [hunk 2]
```diff
@@ -41,23 +41,185 @@
...
+                      "
+                      v-model="line.data.value"
+                      size="sm"
+                      class="mt-3 mb-3"
+                      :placeholder="T.problemCreatorContentPlaceHolder"
+                    />
+                    <b-form-textarea
+                      v-if="
+                        line.data.kind === 'multiline' ||
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue:80]: The computed prop could be used here.

File: frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue [hunk 3]
```diff
@@ -41,23 +41,185 @@
...
+                        v-for="lineoption in lineOptions"
+                        :key="lineoption.kind"
+                        @click="line.data.kind = lineoption.kind"
+                      >
+                        {{ lineoption.type }}
+                      </b-dropdown-item>
+                    </b-dropdown>
+                    <b-button
+                      v-if="
+                        line.data.kind === 'array' ||
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/components/problem/creator/cases/CaseEdit.vue:105]: Here as well.

File: frontend/www/js/omegaup/problem/creator/creator.ts [hunk 4]
```diff
@@ -5,10 +5,17 @@ import { BootstrapVue, BootstrapVueIcons } from 'bootstrap-vue';
...
 
 import 'bootstrap/dist/css/bootstrap.css';
 import 'bootstrap-vue/dist/bootstrap-vue.css';
 
+Vue.directive('Sortable', {
+  inserted: (el: HTMLElement, binding) => {
+    new Sortable(el, binding.value || {});
+  },
+});
+
```
Comment (carlosabcs) [inline frontend/www/js/omegaup/problem/creator/creator.ts:18]: This is not used here anymore.

**Review by carlosabcs — COMMENTED** [review]
Please remove everything related to `sortable` because it's not part of this PR and we should not include code that is not part of what we are implementing.

I suggest you make a copy of this branch before. Then go back to this branch and delete all sortable-related, commit those changes and push them here.

File: frontend/www/js/omegaup/problem/creator/creator.ts (hunk 4 above)
Comment (Aritra8438) [reply frontend/www/js/omegaup/problem/creator/creator.ts]: Removed! @carlosabcs. PTAL!

**Review by carlosabcs — COMMENTED** [review]
Approved.

Can you please give this one more check @pabo99 ?

**Review by heduenas — APPROVED** [review]

**Pull request merged**